datafusion/core/src/execution/context.rs

```diff
@@ -447,6 +447,31 @@ impl SessionContext {
                 }
             }
 
+            LogicalPlan::DropCatalogSchema(DropCatalogSchema {
+                name,
+                if_exists,
+                cascade,
+                ..
+            }) => {
+                let (dereg, schema) = self.find_and_deregister_schema(&name, cascade)?;
+                match (if_exists, schema, dereg) {
+                    // successful deregsiter
+                    (_, _, true) => self.return_empty_dataframe(),
+                    // schema found but failed to deregister
+                    (_, Some(schema), false) => Err(DataFusionError::Execution(format!(
+                        "Cannot drop schema {} because other tables depend on it: {}",
+                        name,
+                        itertools::join(schema.table_names().iter(), ", ")
+                    ))),
+                    // no schema found
+                    (false, None, false) => Err(DataFusionError::Execution(format!(
+                        "Schema '{name}' doesn't exist."
+                    ))),
+                    // no schema found but dont return error
```

Comment:
is it possible scenario?

Reply:
This scenario would occur for a statement like `CREATE SCHEMA does_not_exist`